flexmeasures/data/services/time_series.py

```diff
@@ -315,23 +315,63 @@ def set_bdf_source(bdf: tb.BeliefsDataFrame, source_name: str) -> tb.BeliefsData
 def drop_unchanged_beliefs(bdf: tb.BeliefsDataFrame) -> tb.BeliefsDataFrame:
     """Drop beliefs that are already stored in the database with an earlier belief time.
 
+    Also drop beliefs that are already in the data with an earlier belief time.
+
     Quite useful function to prevent cluttering up your database with beliefs that remain unchanged over time.
-    Only works on BeliefsDataFrames with a unique belief time and unique source.
     """
     if bdf.empty:
         return bdf
-    if len(bdf.lineage.belief_times) > 1:
```

Comment:
Don't we need these checks anymore in the function dealing with the dB data? I saw these assumptions coming back there in the docstring, I believe.